staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -509,6 +459,89 @@ func (g *Cloud) getInstanceByName(name string) (*gceInstance, error) {
 	return nil, cloudprovider.InstanceNotFound
 }
 
+type instanceOrError struct {
+	instance *gceInstance
+	err      error
+}
+
+// Gets the named instances, returning a map of each name to either the found instances or
+// cloudprovider.InstanceNotFound if the instance is not found
+func (g *Cloud) getInstanceOrErrorsByNames(allNames []string) (map[string]*instanceOrError, error) {
+	ctx, cancel := cloud.ContextWithCallTimeout()
+	defer cancel()
+
+	found := map[string]*instanceOrError{}
+	allNamesSet := sets.NewString(allNames...)
+	remaining := len(allNames)
+
+	for i := 0; i < len(allNames); i += 500 {
+		var names []string
+		if i+500 > len(allNames) {
+			names = allNames[i:]
+		} else {
+			names = allNames[i : i+500]
+		}
+		escapedInstanceNames := []string{}
+		for _, name := range names {
+			canonicalizeName := canonicalizeInstanceName(name)
+			escapedInstanceNames = append(escapedInstanceNames, regexp.QuoteMeta(canonicalizeName))
```

Comment:
looks like some things changed in between the large copy-paste of this function and it's impossible to tell what was copy pasted and what is modified. Could you split this into two commits, one for changing the function, the other for moving the function to a new helper function. Right now diffs just show this as a removal of the old function and the addition of a whole new function.

Reply:
I split the commit into two. See if this helps.